chore: remove version from install command in README.md

arctic-alpaca opened the pull request:
<!---
Thanks for creating a pull request!

Please start your PR title with the appropriate prefix as specified by [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#specification):

chore: [your title]
docs: [your title]
feat: [your title]
fix: [your title]
refactor: [your title]
rfc: [your title]
test: [your title]
-->

## Description
<!--- 
For complex PRs, please describe what you did to achieve your goal here.
Did you consider different approaches, where there any trade-offs you made?
More information about why you implemented your solution the way you did helps reviewers.
This section can be removed for obvious changes like typo fixes.
-->
Removes the version from the install command. As @Nemo157 said, this will error on mismatch, but does not provide any benefits.

Nemo157 commented on README.md:
````diff
@@ -27,7 +27,7 @@ Currently only Windows and Linux are supported.
 Install `tricore-probe`:
 ```shell
-cargo install tricore-probe --git https://github.com/veecle/tricore-probe --version 0.2.0
+cargo install tricore-probe --git https://github.com/veecle/tricore-probe --version 0.2.1
````
```suggestion
cargo install tricore-probe --git https://github.com/veecle/tricore-probe
```

I don't think specifying the version is useful when installing from a git repo, I didn't realize cargo would even accept it, I assume it just errors if it mismatches.

arctic-alpaca replied:
I think the initial intention was to allow installing released versions while working on the `main` branch, but that doesn't work anyway.

arctic-alpaca committed:
chore: remove version from install command in README.md

cargo will error on version mismatch, we don't gain anything by specifying the version.
